addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxPlayerHandler.java

```diff
@@ -0,0 +1,432 @@
+	private Logger logger = LoggerFactory
+			.getLogger(SqueezeBoxPlayerHandler.class);
+
+	public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+			.singleton(SQUEEZEBOXPLAYER_THING_TYPE);
+
+	/**
+	 * We need to remember some states to change offsets in volume, time index,
+	 * etc..
+	 */
+	protected Map<String, State> stateMap = Collections
+			.synchronizedMap(new HashMap<String, State>());
+
+	/**
+	 * Keeps current track time
+	 */
+	ScheduledFuture<?> timeCounterJob;
+
+	/**
+	 * Local reference to our bridge
+	 */
+	private SqueezeBoxServerHandler _squeezeBoxServerHandler;
+
+	/**
+	 * Our mac address, needed everywhere
+	 */
+	private String mac;
+
+	/**
+	 * Before we mute or recieve a mute event, store our current volume
+	 */
+	private int unmuteVolume = 0;
+
+	/**
+	 * The server sends us the current time on play/pause/stop events, we
+	 * increment it locally from there on
+	 */
+	private int currentTime = 0;
+
+	/**
+	 * Our we playing something right now or not, need to keep current track
+	 * time
+	 */
+	private boolean playing;
+
+	/**
+	 * Creates SqueezeBox Player Handler
+	 * 
+	 * @param thing
+	 */
+	public SqueezeBoxPlayerHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void initialize() {
+		mac = getThing().getConfiguration().as(SqueezeBoxPlayerConfig.class).mac;
+		timeCounter();
+		if (squeezeBoxServerHandler() != null) {
+			updateStatus(
+					squeezeBoxServerHandler().getThing().getStatus());
+		}
+	};
+
+	@Override
+	public void dispose() {
+
+		// stop our duration counter
+		if (timeCounterJob != null && !timeCounterJob.isCancelled()) {
+			timeCounterJob.cancel(true);
+			timeCounterJob = null;
+		}
+
+		if (squeezeBoxServerHandler() != null) {
+			squeezeBoxServerHandler().removePlayerCache(mac);
+		}
+		logger.debug("Thing {} disposed.", getThing().getUID());
+		super.dispose();
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		if (squeezeBoxServerHandler() == null) {
+			logger.warn("Player has no server configured, ignoring command");
+			return;
+		}
+		String mac = getConfigAs(SqueezeBoxPlayerConfig.class).mac;
+		switch (channelUID.getIdWithoutGroup()) {
+		case CHANNEL_POWER:
+			if (command.equals(OnOffType.ON)) {
+				squeezeBoxServerHandler().powerOn(mac);
+			} else {
+				squeezeBoxServerHandler().powerOff(mac);
+			}
+			break;
+		case CHANNEL_MUTE:
+			if (command.equals(OnOffType.ON)) {
+				mute();
+			} else {
+				squeezeBoxServerHandler().unMute(mac, unmuteVolume);
+			}
+			break;
+		case CHANNEL_STOP:
+			if (command.equals(OnOffType.ON)) {
+				squeezeBoxServerHandler().stop(mac);
+			} else {
+				squeezeBoxServerHandler().play(mac);
+			}
+			break;
+		case CHANNEL_VOLUME:
+			if (command instanceof PercentType) {
+				squeezeBoxServerHandler().setVolume(mac,
+						((PercentType) command).intValue());
+			} else if (command.equals(IncreaseDecreaseType.INCREASE)
+					|| command.equals(UpDownType.UP)) {
```

Comment:
UP and DOWN don't need to be supported here - the volume is always a Dimmer Item, which only supports INCREASE/DECREASE.